Bottom: bb543769bcc081e75e62da5e3d8d22498c38d03b
Top:    2b68c056e93d81dd85bd74a2b66677376f07c711
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 16:59:14 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 37aaebb..2df8cd9 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
